sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -1377,29 +1389,59 @@ object GpuOverrides {
         override def convertToGpu(lhs: Expression, rhs: Expression): GpuExpression =
           GpuAdd(lhs, rhs)
       }),
     expr[Subtract](
       "Subtraction",
       ExprChecks.binaryProjectNotLambda(
-        TypeSig.integral + TypeSig.fp, TypeSig.numericAndInterval,
-        ("lhs", TypeSig.integral + TypeSig.fp, TypeSig.numericAndInterval),
-        ("rhs", TypeSig.integral + TypeSig.fp, TypeSig.numericAndInterval)),
+        TypeSig.numeric, TypeSig.numericAndInterval,
+        ("lhs", TypeSig.numeric, TypeSig.numericAndInterval),
+        ("rhs", TypeSig.numeric, TypeSig.numericAndInterval)),
       (a, conf, p, r) => new BinaryExprMeta[Subtract](a, conf, p, r) {
         override def convertToGpu(lhs: Expression, rhs: Expression): GpuExpression =
           GpuSubtract(lhs, rhs)
       }),
     expr[Multiply](
       "Multiplication",
-      ExprChecks.binaryProjectNotLambda(TypeSig.integral + TypeSig.fp, TypeSig.numeric,
-        ("lhs", TypeSig.integral + TypeSig.fp, TypeSig.numeric),
-        ("rhs", TypeSig.integral + TypeSig.fp, TypeSig.numeric)),
+      ExprChecks.binaryProjectNotLambda(TypeSig.numeric +
+          TypeSig.psNote(TypeEnum.DECIMAL, "Because of Spark's inner workings the full range " +
+              "of decimal precision (even for 64-bit value) is not supported."),
+        TypeSig.numeric,
+        ("lhs", TypeSig.numeric, TypeSig.numeric),
+        ("rhs", TypeSig.numeric, TypeSig.numeric)),
       (a, conf, p, r) => new BinaryExprMeta[Multiply](a, conf, p, r) {
+        override def tagExprForGpu(): Unit = {
+          // Multiplication of Decimal types is a little odd. Spark will cast the inputs
+          // to a common wider value where scale is max of the two input scales, and precision is
+          // max of the two input non-scale portions + the new scale). Then it will do the multiply,
+          // which will produce a return scale that is 2x that of the wider scale, but lie about it
+          // in the return type of the Multiply operator. Then in CheckOverflow it will reset the
+          // scale and check the precision so that they know it fits in final desired result which
+          // is precision1 + precision2 + 1 for precision and scale1 + scale2 for scale, based off
+          // of the precision and scale for the original input values. We would like to avoid all
+          // of this if possible because having a temporary intermediate value that can have a
+          // scale quite a bit larger than the final result reduces the maximum precision that
+          // we could support, as we don't have unlimited precision. But sadly because of how
+          // the logical plan is compiled down to the physical plan we have lost what the original
+          // types were and cannot recover it. As such for now we are going to do what spark does
+          // but we have to recompute/recheck the temporary precision to be sure it will fit
+          // on the GPU.
```

Comment:
Shoudl there be a tracking issue for this problem that's referenced here?  Seems like an issue we'd want to track, even if it's just a place to record ideas to make it better in the future for now.  It also can hold the details of the decimal precision issue in the description, and we could reference the github issue in the PS note if the reader wants more details.

Reply:
https://github.com/NVIDIA/spark-rapids/issues/1571